pkg/storage/stores/shipper/indexshipper/tsdb/index/chunk_test.go

```diff
@@ -145,6 +145,104 @@ func TestChunkMetasFinalize(t *testing.T) {
 	}
 }
 
+func TestChunkMetasStats(t *testing.T) {
+	for _, tc := range []struct {
+		desc          string
+		chks          []ChunkMeta
+		from, through int64
+		exp           ChunkStats
+	}{
+		{
+			desc: "full range",
+			chks: []ChunkMeta{
+				{MinTime: 0, MaxTime: 1, KB: 1, Entries: 1},
+				{MinTime: 1, MaxTime: 2, KB: 1, Entries: 1},
+				{MinTime: 2, MaxTime: 3, KB: 1, Entries: 1},
+				{MinTime: 3, MaxTime: 4, KB: 1, Entries: 1},
+				{MinTime: 4, MaxTime: 5, KB: 1, Entries: 1},
+			},
+			from:    0,
+			through: 5,
+			exp: ChunkStats{
+				Chunks:  5,
+				KB:      5,
+				Entries: 5,
+			},
+		},
+		{
+			desc: "overlapping",
+			chks: []ChunkMeta{
+				{MinTime: 0, MaxTime: 1, KB: 1, Entries: 1},
+				{MinTime: 1, MaxTime: 4, KB: 1, Entries: 1},
+				{MinTime: 2, MaxTime: 3, KB: 1, Entries: 1},
+				{MinTime: 3, MaxTime: 5, KB: 2, Entries: 2},
+				{MinTime: 4, MaxTime: 5, KB: 1, Entries: 1},
+			},
+			from:    0,
+			through: 5,
+			exp: ChunkStats{
```

Comment:
I think this testcase is an amazing representation of our current problem.
Let me try to do a different math:
I only take the following chunks (subset of the test's data):
{MinTime: 0, MaxTime: 1, KB: 1, Entries: 1},
{MinTime: 2, MaxTime: 3, KB: 1, Entries: 1}
{MinTime: 3, MaxTime: 5, KB: 2, Entries: 2}
result would be: {Chunks:3, KB: 4, Entries: 4}

If I take a closer look, all I can tell is we have 2 possible duplicates, so 6-2 would end up with 4. What is happening here?

Reply:
What's with `{MinTime: 1, MaxTime: 4, KB: 1, Entries: 1}` how would we get that out?